jarviscli/plugins/memory.py

```diff
@@ -0,0 +1,75 @@
+from plugin import plugin
+import time
+import random
+
+
+@plugin('memory')
+class Memory:
+    """
+    Welcome to the Short-Term Memory Trainer! Here you can find
+    all the functionalities of this plugin.
+    Usage: Type memory, press enter and follow the instructions
+    Functionalities: You can train your memory by recalling a
+    random generated number.
+    Find out how many digits you can remember.
+    """
+
+    def __call__(self, jarvis, s):
+        jarvis.say("Welcome to Memory Trainer.")
+        jarvis.say("We will train you by giving a number to remember.")
+        jarvis.say("The number won't be visible for very long.")
+        jarvis.say("Each time the number will get 1 more digit.")
+        answer = self.get_answer(jarvis)
+        if answer == 1:
+            self.trainer(jarvis)
+        else:
+            return
+
+    def get_answer(self, jarvis):
+        """
+        Asks the user you check if he is ready to start the game
+        1 for ready, 2 for not ready
+        """
+        jarvis.say("Are you ready to play?")
+        while True:
+            try:
+                jarvis.say("Press 1 for Yes. \nPress 2 for No.")
+                c = int(input("Your choice: "))
+                if c != 1 or c != 2:
+                    raise ValueError('Please give a valid input')
+                break
+            except ValueError:
+                print("Please give a valid input")
+        return c
+
+    def trainer(self, jarvis):
+        """
+        User guesses the number, checks if the guess is correct.
+        First wrong guess means the user has lost.
+        """
+        lost = False
+        number = ""
+        while not lost:
+            number = self.create_number(jarvis, number)
+            print("The number will only be visible for a few seconds.")
+            print(number, end="\r")
```

Comment:
Does not work at last for some terminals. '\r' does delete complete line on my terminal (kitty on linux).

For better solution see ``utilities/animations.py ``:

* Do not print '\r' here
* Print '\r' before 'Time ended' below
* Call sys.stdout.flush() after this line so this line is printed even without newline.